sklearn/cluster/k_means_.py

```diff
@@ -170,6 +170,9 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
             algorithm="auto", return_n_iter=False):
     """K-means clustering algorithm.
 
+    Warning: The data will be converted to C ordering, which might cause a
```

Comment:
I would avoid to issue a warning here. You can add a line of explanation under X, similarly to the some explanation regarding sparse matrix [CSR](http://scikit-learn.org/dev/modules/generated/sklearn.preprocessing.QuantileTransformer.html#sklearn.preprocessing.QuantileTransformer.fit)

Also a conversion will for sure make a memory copy so you should remove "might" :)